src/test/java/org/opentripplanner/routing/fares/OrcaFareServiceTest.java

```diff
@@ -0,0 +1,199 @@
+     * make direct calls to the Orca fare service with predefined routes. Where the default fare is applied a test
+     * substitute {@link OrcaFareServiceImpl#DEFAULT_TEST_RIDE_PRICE} is used. This will be the same for all cash fare
+     * types.
+     */
+    @ParameterizedTest
+    @MethodSource("trips")
+    public void calculateFareTest(List<Ride> rides,
+                                  Fare.FareType fareType,
+                                  float expectedFareInCents
+    ) {
+        Fare fare = new Fare();
+        orcaFareService.calculateFare(fare, null, fareType, rides, null);
+        Assertions.assertEquals(expectedFareInCents, fare.getFare(fareType).getCents());
+    }
+
+    private static Stream<Arguments> trips() {
+        List<Ride> trip1 = getTrip(
+            COMM_TRANS_COMMUTER_EXPRESS
+        );
+        List<Ride> trip2 = getTrip(
+            KITSAP_TRANSIT,
+            WASHINGTON_STATE_FERRIES,
+            COMM_TRANS_LOCAL_SWIFT
+        );
+        List<Ride> trip3 = getTrip(
+            PIERCE_COUNTY_TRANSIT,
+            SOUND_TRANSIT,
+            COMM_TRANS_COMMUTER_EXPRESS
+        );
+        List<Ride> trip4 = getTrip(
+            EVERETT_TRANSIT,
+            COMM_TRANS_COMMUTER_EXPRESS,
+            SOUND_TRANSIT
+        );
+        List<Ride> trip5 = getTrip(
+            KC_WATER_TAXI_VASHON_ISLAND,
+            KC_METRO
+        );
+        List<Ride> trip6 = getTrip(
+            SEATTLE_STREET_CAR,
+            KC_METRO,
+            COMM_TRANS_COMMUTER_EXPRESS
+        );
+        return Stream.of(
+            Arguments.of(trip1, Fare.FareType.regular, DEFAULT_RIDE_PRICE_IN_CENTS),
+            Arguments.of(trip1, Fare.FareType.senior, DEFAULT_RIDE_PRICE_IN_CENTS),
+            Arguments.of(trip1, Fare.FareType.youth, DEFAULT_RIDE_PRICE_IN_CENTS),
+            Arguments.of(trip1, Fare.FareType.orcaLift, 200f),
+            Arguments.of(trip1, Fare.FareType.orcaRegular, DEFAULT_RIDE_PRICE_IN_CENTS),
+            Arguments.of(trip1, Fare.FareType.orcaSenior, 200f),
+            Arguments.of(trip1, Fare.FareType.orcaYouth, 300f),
+            // The following costs are made up of the expected cost to be applied for each agency. This is usually the
+            // most expensive leg of the journey.
+            Arguments.of(trip2, Fare.FareType.orcaLift, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 125f),
+            Arguments.of(trip2, Fare.FareType.orcaRegular, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + DEFAULT_RIDE_PRICE_IN_CENTS),
+            Arguments.of(trip2, Fare.FareType.orcaSenior, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 125f),
+            Arguments.of(trip2, Fare.FareType.orcaYouth, 200f + DEFAULT_RIDE_PRICE_IN_CENTS + 0f),
+            Arguments.of(trip3, Fare.FareType.orcaLift, DEFAULT_RIDE_PRICE_IN_CENTS + 0f + 0f),
+            Arguments.of(trip3, Fare.FareType.orcaRegular, 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 0f),
```

Comment:
It'd be helpful to have comments about each of these test cases. It's not immediately clear why the first of these has the `DEFAULT_RIDE_PRICE_IN_CENTS` as the first variable in the equation, whereas the second has it as the second and yet they still would add up to the same value.

Reply:
I've added a general comment as the approach is the same for all.